examples/hitl/rearrange_v2/rearrange_v2.py

```diff
@@ -365,11 +367,32 @@ def pop_ui_events(self) -> List[Dict[str, Any]]:
         return events
 
     def _get_camera_lookat_pos(self) -> mn.Vector3:
+        # HACK: Estimate camera height.
+        # TODO: GuiController should have knowledge on the agent type it controls, and should provide an API to get the camera height.
+        gui_agent_controller = self.gui_agent_controller
+        sim = self.app_service.sim
+        agent_data = self.agent_data
+        camera_height_offset = 1.0
+        if isinstance(gui_agent_controller, GuiRobotController):
+            # The robot camera height is defined in `spot_robot.py`.
+            # It's directly attached to the 6th articulated object link, without a height offset.
+            agent = sim.agents_mgr[agent_data.agent_index].articulated_agent
+            root = agent.sim_obj
+            head_link = root.get_link_scene_node(6)
+            head_link_height = head_link.absolute_translation.y
+            base_link = root.get_link_scene_node(-1)
+            base_link_height = base_link.absolute_translation.y
+            camera_height_offset = head_link_height - base_link_height
+        elif isinstance(gui_agent_controller, GuiHumanoidController):
```

Comment:
I think I can just expand this condition to trigger when `gui_agent_controller` is `LLMController`

Reply:
Correct!
Eventually, the controller will know about the agent it's controlling and will be able to get the correct camera height.
Better have this hack in and expose the complexity for now.